packages/tools/examples/videoContourSegmentation/index.ts

```diff
@@ -0,0 +1,484 @@
+
+  toolGroup.addTool(SegmentationDisplayTool.toolName);
+  toolGroup.addTool(SplineContourSegmentationTool.toolName);
+  toolGroup.addTool(LivewireContourSegmentationTool.toolName);
+  toolGroup.addTool(PlanarFreehandContourSegmentationTool.toolName);
+  toolGroup.addTool(LivewireContourSegmentationTool.toolName);
+
+  toolGroup.addToolInstance(
+    'CatmullRomSplineROI',
+    SplineContourSegmentationTool.toolName,
+    {
+      spline: {
+        type: SplineContourSegmentationTool.SplineTypes.CatmullRom,
+      },
+    }
+  );
+
+  toolGroup.addToolInstance(
+    'LinearSplineROI',
+    SplineContourSegmentationTool.toolName,
+    {
+      spline: {
+        type: SplineContourSegmentationTool.SplineTypes.Linear,
+      },
+    }
+  );
+
+  toolGroup.addToolInstance(
+    'BSplineROI',
+    SplineContourSegmentationTool.toolName,
+    {
+      spline: {
+        type: SplineContourSegmentationTool.SplineTypes.BSpline,
+      },
+    }
+  );
+
+  toolGroup.setToolEnabled(SegmentationDisplayTool.toolName);
+
+  toolGroup.setToolActive(splineToolsNames[0], {
+    bindings: [
+      {
+        mouseButton: MouseBindings.Primary, // Left Click
+      },
+    ],
+  });
+
+  // Get Cornerstone imageIds and fetch metadata into RAM
+  const imageIds = await createImageIdsAndCacheMetaData({
+    StudyInstanceUID: '2.25.96975534054447904995905761963464388233',
+    SeriesInstanceUID: '2.25.15054212212536476297201250326674987992',
+    wadoRsRoot:
+      getLocalUrl() || 'https://d33do7qe4w26qo.cloudfront.net/dicomweb',
+  });
+
+  // Only one SOP instances is DICOM, so find it
+  const videoId = imageIds.find(
+    (it) => it.indexOf('2.25.179478223177027022014772769075050874231') !== -1
+  );
+
+  // Instantiate a rendering engine
+  const renderingEngineId = 'myRenderingEngine';
+  const renderingEngine = new RenderingEngine(renderingEngineId);
+
+  // Create the viewports
+  const viewportInputArray = [
+    {
+      viewportId: viewportId,
+      type: ViewportType.VIDEO,
+      element: element,
+      defaultOptions: {
+        background: <Types.Point3>[0.2, 0, 0.2],
+      },
+    },
+  ];
+
+  renderingEngine.setViewports(viewportInputArray);
+  toolGroup.addViewport(viewportId, renderingEngineId);
+
+  // Get the stack viewport that was created
+  viewport = <Types.IStackViewport>renderingEngine.getViewport(viewportId);
+
+  // Set the stack on the viewport
+  viewport.setVideo(videoId, 1);
+
+  // Render the image
+  renderingEngine.render();
+  const seconds = (time) => `${Math.round(time * 10) / 10} s`;
+
+  element.addEventListener(Enums.Events.IMAGE_RENDERED, (evt: any) => {
+    const { time, duration } = evt.detail;
+    rangeElement.value = time;
+    rangeElement.max = duration;
+    const timeElement = document.getElementById('time');
+    timeElement.innerText = seconds(time);
+    const remainingElement = document.getElementById('remaining');
+    remainingElement.innerText = seconds(duration - time);
+  });
```

Comment:
can you refactor this logic outside, it is out of place IMO

Reply:
It is GUI logic, not part of the viewport itself, so I'm not sure where it would go if not here.  It really doesn't belong in CS3D, but if you have a good place to put all that logic, like maybe a helper method, I could do so.